keanu-project/src/main/java/io/improbable/keanu/distributions/continuous/Gamma.java

```diff
@@ -18,123 +18,124 @@
-            throw new IllegalArgumentException("Invalid value for b");
-        }
-        return location - lambda * Math.log(random.nextDouble());
+    private static double exponentialSample(double location, double alpha, KeanuRandom random) {
+        return location - alpha * Math.log(random.nextDouble());
     }
 
     @Override
     public DoubleTensor logProb(DoubleTensor x) {
-        final DoubleTensor aMinusXOverTheta = location.minus(x).divInPlace(theta);
-        final DoubleTensor kLnTheta = k.times(theta.log());
-        final DoubleTensor xMinusAPowKMinus1 = x.minus(location).powInPlace(k.minus(1));
-        final DoubleTensor lnXMinusAToKMinus1 = ((xMinusAPowKMinus1).divInPlace(k.apply(org.apache.commons.math3.special.Gamma::gamma))).logInPlace();
-        return aMinusXOverTheta.minusInPlace(kLnTheta).plusInPlace(lnXMinusAToKMinus1);
+        final DoubleTensor aMinusXOverScale = location.minus(x).divInPlace(scale);
+        final DoubleTensor alphaLnScale = alpha.times(scale.log());
+        final DoubleTensor xMinusAPowAlphaMinus1 = x.minus(location).powInPlace(alpha.minus(1.));
+        final DoubleTensor lnXMinusAToAlphaMinus1 = ((xMinusAPowAlphaMinus1).divInPlace(alpha.apply(org.apache.commons.math3.special.Gamma::gamma))).logInPlace();
+        return aMinusXOverScale.minusInPlace(alphaLnScale).plusInPlace(lnXMinusAToAlphaMinus1);
     }
 
     @Override
     public Diffs dLogProb(DoubleTensor x) {
         final DoubleTensor xMinusLocation = x.minus(location);
         final DoubleTensor locationMinusX = location.minus(x);
-        final DoubleTensor kMinus1 = k.minus(1.);
-        final DoubleTensor oneOverTheta = theta.reciprocal();
+        final DoubleTensor alphaMinus1 = alpha.minus(1.);
+        final DoubleTensor oneOverScale = scale.reciprocal();
 
-        final DoubleTensor dLogPdx = kMinus1.div(xMinusLocation).minusInPlace(oneOverTheta);
-        final DoubleTensor dLogPdlocation = kMinus1.div(locationMinusX).plusInPlace(oneOverTheta);
-        final DoubleTensor dLogPdtheta = theta.times(k).plus(locationMinusX).divInPlace(theta.pow(2.)).unaryMinusInPlace();
-        final DoubleTensor dLogPdk = xMinusLocation.logInPlace().minusInPlace(theta.log()).minusInPlace(k.apply(org.apache.commons.math3.special.Gamma::digamma));
+        final DoubleTensor dLogPdlocation = alphaMinus1.div(locationMinusX).plusInPlace(oneOverScale);
+        final DoubleTensor dLogPdscale = scale.times(alpha).plus(locationMinusX).divInPlace(scale.pow(2.)).unaryMinusInPlace();
+        final DoubleTensor dLogPdalpha = xMinusLocation.logInPlace().minusInPlace(scale.log()).minusInPlace(alpha.apply(org.apache.commons.math3.special.Gamma::digamma));
+        final DoubleTensor dLogPdx = alphaMinus1.div(xMinusLocation).minusInPlace(oneOverScale);
 
         return new Diffs()
         .put(A, dLogPdlocation)
-        .put(THETA, dLogPdtheta)
-        .put(K, dLogPdk)
+        .put(THETA, dLogPdscale)
+        .put(K, dLogPdalpha)
```

Comment:
if we go with `alpha` then you should also change `K` to `ALPHA`

Reply:
I'm going to do this work after we decide on the parameter names.